geomesa-arrow/geomesa-arrow-datastore/src/main/scala/org/locationtech/geomesa/arrow/data/ArrowDataStore.scala

```diff
@@ -32,19 +33,19 @@ import scala.util.control.NonFatal
 class ArrowDataStore(val url: URL, caching: Boolean) extends ContentDataStore with FileDataStore
     with HasGeoMesaMetadata[String] with HasGeoMesaStats {
 
-  import org.locationtech.geomesa.arrow.allocator
-
   private var initialized = false
 
   // note: to avoid cache issues, don't allow writing if caching is enabled
   private lazy val writable = !caching && Try(createOutputStream()).map(_.close()).isSuccess
 
+  private lazy val allocator = ArrowAllocator("arrow-datastore")
```

Comment:
Ok, time to ask lots of silly questions...

What happens if one creates multiple Arrow DSes in the same JVM?  Does the apply method here create separate allocators?  Or is there a singleton per string?

Reply:
the name is just used for debugging, but it's not a unique key or anything. each call to apply will create a new child allocator, regardless of the name. I'll update the scaladocs to indicate that.